File: problem_15/problem_15.js
const now = require("performance-now");

const problem15 = (m, n, memo={}) => {
    if (m === 1 || n === 1) { return 1 }
    const key = `${m},${n}`;
    if (memo.hasOwnProperty(key)){
        return memo[key];
    }
    else {
        let result = problem15(m-1, n, memo) + problem15(m, n-1, memo);
        memo[key] = result;
        return result;
    }   
}

let start = now()
let solution = problem15(21,21)
console.log(`${solution} found in ${(now()-start) / 1000} seconds.`)
// 137846528820 found in 0.000742721000000003 seconds.

File: problem_15/problem_15.py
import time

def problem_15(m, n, memo = {}):
    if m == 1 or n == 1:
        return 1
    try:
        return memo[(m,n)]
    except KeyError:
        result = problem_15(m-1, n, memo) + problem_15(m, n-1, memo)
        memo[(m,n)] = result
        return result

start = time.time()
solution = problem_15(21,21)
print(f"{solution} found in {time.time() - start} seconds.")
# 137846528820 found in 0.0006380081176757812 seconds.